src/test/run-make-fulldeps/crate-hash-rustc-version/Makefile

```diff
@@ -0,0 +1,32 @@
+-include ../../run-make-fulldeps/tools.mk
+
+# Ensure that crates compiled with different rustc versions cannot
+# be dynamically linked.
+
+FLAGS := -Cprefer-dynamic -Zsymbol-mangling-version=v0
+UNAME := $(shell uname)
+ifeq ($(UNAME),Linux)
+  EXT=".so"
+  NM_CMD := nm -D
+endif
+ifeq ($(UNAME),Darwin)
+  EXT=".dylib"
+  NM_CMD := nm
+endif
+
+all:
+	# a.rs is a dylib
+	$(RUSTC) a.rs --crate-type=dylib $(FLAGS)
+	# Write symbols to disk.
+	$(NM_CMD) $(call DYLIB,a) > $(TMPDIR)/symbolsbefore
```

Comment:
This test will need to be disabled if `NM_CMD` is not set I think.

Reply:
What's the best way to do this? Just exit on the first line?

```
if [ -n "$NM_CMD" ]; then echo "`nm` not available"; exit 0; fi
```

Reply:
@bjorn3 https://github.com/rust-lang/rust/commit/87f2d6a24389b4a13e3bb4a539aa07847c0b1346#diff-6cfd458f9e222a14dcedbbf54dc9ed6e9fec50623a530fbe310437a9d61b091aR18

Reply:
I believe make runs every line in it's own shell, so that would only exit that specific line. I think you can do something like

```make
ifndef NM_CMD
all:
    exit 0
else
all:
    # the test code...
endif